node/src/components/block_accumulator.rs

```diff
@@ -280,11 +326,12 @@ impl BlockAccumulator {
             debug!(%block_hash, "ignoring outdated block");
             return Effects::new();
         }
+        self.upsert_acceptor(*block_hash, Some(era_id), sender);
 
-        let acceptor = self
-            .block_acceptors
-            .entry(*block_hash)
-            .or_insert_with(|| BlockAcceptor::new(*block_hash, None));
+        let acceptor = match self.block_acceptors.get_mut(block_hash) {
+            None => return Effects::new(),
```

Comment:
Would it be better to have `upsert_acceptor` return an `Option<&mut BlockAcceptor>` (or even the filled entry) to avoid this kind of construct? If not, can we at least but an explanation in a comment here?

Reply:
I tried, but it doesn't work with the borrow checker, because then all of `self` is borrowed mutably.
I'll add a comment.

Reply:
Right, we're missing partial borrows here. There's no way to convey (at the moment) that the returned reference only borrows one field of the struct mutably, instead of the entire thing. :(